plasma_framework/contracts/src/utils/RLP.sol

```diff
@@ -0,0 +1,605 @@
+pragma solidity ^0.5.0;
+
+
+/**
+ * @title RLP
+ * @dev Library for RLP decoding.
+ * Based off of https://github.com/androlo/standard-contracts/blob/master/contracts/src/codec/RLP.sol.
+ */
+library RLP {
+    /*
+     * Storage
+     */
+
+    uint internal constant DATA_SHORT_START = 0x80;
+    uint internal constant DATA_LONG_START = 0xB8;
+    uint internal constant LIST_SHORT_START = 0xC0;
+    uint internal constant LIST_LONG_START = 0xF8;
+
+    uint internal constant DATA_LONG_OFFSET = 0xB7;
+    uint internal constant LIST_LONG_OFFSET = 0xF7;
+
+    struct RLPItem {
+        uint _unsafe_memPtr;    // Pointer to the RLP-encoded bytes.
+        uint _unsafe_length;    // Number of bytes. This is the full length of the string.
+    }
+
+    struct Iterator {
+        RLPItem _unsafe_item;   // Item that's being iterated over.
+        uint _unsafe_nextPtr;   // Position of the next item in the list.
+    }
+
+
+    /*
+     * Internal functions
+     */
+
+    /**
+     * @dev Creates an RLPItem from an array of RLP encoded bytes.
+     * @param self The RLP encoded bytes.
+     * @return An RLPItem.
+     */
+    function toRLPItem(bytes memory self)
+        internal
+        pure
+        returns (RLPItem memory)
+    {
+        uint len = self.length;
+        if (len == 0) {
+            return RLPItem(0, 0);
+        }
+        uint memPtr;
+        assembly {
+            memPtr := add(self, 0x20)
+        }
+        return RLPItem(memPtr, len);
+    }
+
+    /**
+     * @dev Creates an RLPItem from an array of RLP encoded bytes.
+     * @param self The RLP encoded bytes.
+     * @param strict Will throw if the data is not RLP encoded.
+     * @return An RLPItem
+     */
+    function toRLPItem(bytes memory self, bool strict)
+        internal
+        pure
+        returns (RLPItem memory)
+    {
+        RLPItem memory item = toRLPItem(self);
+        if (strict) {
+            uint len = self.length;
+            if (_payloadOffset(item) > len) {
+                revert();
```

Comment:
my IDE is complaining all those `revert()` should have message.
I think we can either pass in a message for them or we just use `require`. Don't have specific opinion.

Reply:
done